website/demos/ContextMenu.tsx

```diff
@@ -129,30 +52,32 @@ function rowKeyGetter(row: Row) {
   return row.id;
 }
 
-function rowRenderer(key: React.Key, props: RowRendererProps<Row>) {
-  return (
-    // @ts-expect-error
-    <ContextMenuTrigger key={key} id="grid-context-menu" collect={() => ({ rowIdx: props.rowIdx })}>
-      <GridRow {...props} />
-    </ContextMenuTrigger>
-  );
-}
-
 export default function ContextMenuDemo({ direction }: Props) {
   const [rows, setRows] = useState(createRows);
   const [nextId, setNextId] = useReducer((id: number) => id + 1, rows[rows.length - 1].id + 1);
+  const [contextMenuProps, setContextMenuProps] = useState<{
+    rowIdx: number;
+    top: number;
+    left: number;
+  } | null>(null);
+  const menuRef = useRef<HTMLMenuElement | null>(null);
+
+  useLayoutEffect(() => {
+    if (contextMenuProps === null) return;
+
+    function onClick(event: MouseEvent) {
+      if (event.target instanceof Node && menuRef.current?.contains(event.target)) {
+        return;
+      }
+      setContextMenuProps(null);
+    }
 
-  function onRowDelete(e: React.MouseEvent<HTMLDivElement>, { rowIdx }: { rowIdx: number }) {
-    setRows([...rows.slice(0, rowIdx), ...rows.slice(rowIdx + 1)]);
-  }
+    addEventListener('click', onClick);
 
-  function onRowInsertAbove(e: React.MouseEvent<HTMLDivElement>, { rowIdx }: { rowIdx: number }) {
-    insertRow(rowIdx);
-  }
-
-  function onRowInsertBelow(e: React.MouseEvent<HTMLDivElement>, { rowIdx }: { rowIdx: number }) {
-    insertRow(rowIdx + 1);
-  }
+    return () => {
+      removeEventListener('click', onClick);
+    };
+  }, [contextMenuProps]);
```

Comment:
Rather than depending on the object itself which could be updated easily, we could have `const isContextMenuOpen = contextMenuProps !== null` outside the effect and use it